site/src/lib/api-client.ts

```diff
@@ -44,19 +48,21 @@ const axiosClient = axios.create({
 });
 
 export type ApiClientError = AxiosError<{
-  errors?: Partial<ValidationError>[];
+  errors?: Partial<ValidationError & { code?: string }>[];
```

Comment:
not sure if this is the best way of doing this. we use `error.code` to have predefined error types, on top of `error.message`. Because we need the logic for `depending on the type of the error, show the error message next to the related text field`. Messages could change, they're not reliable. Code's will not change unless we change the logic of BE.